src/execution_tree/primitives/sub_operation.cpp

```diff
@@ -93,37 +133,16 @@ namespace phylanx { namespace execution_tree { namespace primitives
 
         using array_type = Eigen::Array<double, Eigen::Dynamic, 1>;
         using result_type = Eigen::Matrix<double, Eigen::Dynamic, 1>;
-        using array_map_type = Eigen::Map<array_type const>;
 
-        if(ops.size()==2)
-        {
-            auto lambda = [&]() -> result_type {
-                auto val1 = *ops.begin();
-                auto val2 = *(++ops.begin());
-                array_map_type first(val1.data(), lhs_size);
-                array_map_type second(val2.data(), lhs_size);
-                return first - second;
-            };
-            result_type result = lambda();
-            return ir::node_data<double>(std::move(result));
-        }
-        else
-        {
-            auto val1 = *ops.begin();
-            array_map_type first(val1.data(), lhs_size);
-            Eigen::Matrix<double, Eigen::Dynamic, 1> result =
-                    std::accumulate(ops.begin()+1, ops.end(),
-                                    array_type{first},
-                                    [&](array_type const& result, ir::node_data<double> const& curr)
-                                            ->  array_type
+        array_type first = ops.begin()->matrix().array();
+        result_type result = std::accumulate(ops.begin()+1, ops.end(),
+                                    first,
+                                    [&](array_type& result, ir::node_data<double> const& curr) ->  array_type
```

Comment:
Please keep the line length under 90 characters, also using clang-format helps maintaining formatting consistency.